src/features/device/deviceSlice.ts

```diff
@@ -52,39 +50,35 @@ export const deviceSlice = createSlice({
       if (action.payload) {
         // If whatever is passed in when this is called is not null
         state.activeWaypoint = null;
-        state.showAlgosAccordion = false;
+        state.infoPane = null;
       }
     },
 
-    setActiveWaypoint: (
-      state,
-      action: PayloadAction<Waypoint["id"] | null>
-    ) => {
+    setActiveWaypoint: (state, action: PayloadAction<Waypoint | null>) => {
       state.activeWaypoint = action.payload;
       if (action.payload) {
-        // If whatever is passed in when this is called is not null; want only one active of either node, wp, or accordion
+        // If there is an active waypoint then we don't want another info screen
+        state.infoPane = "waypoint";
         state.activeNode = null;
-        state.showAlgosAccordion = false;
-        // Only want edit to be true if someone explicitly clicks on edit; prevent holdover from exiting previous node exit
-        state.waypointEdit = false;
       }
     },
 
-    setWaypointEdit: (state, action: PayloadAction<boolean>) => {
-      state.waypointEdit = action.payload;
-    },
-
-    setAllowOnMapWaypointCreation: (state, action: PayloadAction<boolean>) => {
-      state.allowOnMapWaypointCreation = action.payload;
-    },
+    setInfoPane: (
+      state,
+      action: PayloadAction<"waypoint" | "waypointEdit" | "algos" | null>
+    ) => {
+      state.infoPane = action.payload;
 
-    setShowAlgosAccordion: (state, action: PayloadAction<boolean>) => {
-      state.showAlgosAccordion = action.payload;
       if (action.payload) {
```

Comment:
Not sure I completely understand - 

I think we should only have one of activeWaypoint or activeNode as non-null at a time. 

If the `action.payload` is truthy (i.e. something waypoints related) then we set `activeNode=null` to make sure to prevent the info pane for `activeNode` from popping up. 

If it's falsy, then we set `activeWaypoint=null`, and the user can click a node or do something that would make `activeNode` non-null.

Reply:
I guess my point of confusion was around this flow not including the `activeNode` code, since we close the node info pane by setting `activeNode` to `null`. Do you have thoughts on whether it would be possible/worth it to include the `activeNode` flow in this method?

Reply:
I think it would be nice to do it, but it might be better as its own PR, since the node pane is currently connected to `activeNode`, and we might be changing the type of `activeNode` to include waypoints.

Reply:
Yeah that's fair, good to resolve this in that case